### PR TITLE
empty extension of Google drive file should not cause dot to be appended to file name

### LINKED ISSUE #37044: ownCloud should not respond 40x for Google Drive files without an extension
### Steps to reproduce
1. In **Google Drive**, upload any file without an extension ("table_of_contents" from an mbox folder for example)
1. Add a Google Drive external storage
1. Within **ownCloud web UI**, navigate to Google Drive external storage and find the file without an extension
    _AR: note the filename has a dangling dot: "table_of_contents."_
1. Click the "table_of_contents." file to start a download
    _ER: download prompt should appear and download should proceed_
    _AR: 40x error page_

### Expected behaviour
download prompt should appear and download should proceed

### Actual behaviour
40x error page

### Server configuration
**Operating system**: Ubuntu 18.04.4 LTS

**Web server:** Apache 2.4.29

**Database:**: MySQL 5.7.29

**PHP version:** 7.2.24

**ownCloud version:** 10.3.2

**Updated from an older ownCloud or fresh install:** fresh install

**Where did you install ownCloud from:** zip from owncloud.org

**Signing status (ownCloud 9.0 and above):** 
```
No errors have been found.
```

**Are you using external storage, if yes which one:** Google Drive

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** None

### Client configuration
**Browser:** Chrome latest, Safari, macOS client

**Operating system:** macOS 10.14.6

### PR DESCRIPTION
**Pull request by phil-davis**
## Description
This is a squashed version of PR #37045 with the changelog edited to have a shorter description.
See that PR for the details and discussion.

## Related Issue
- Fixes #37044 

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [x] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by phil-davis** [comment]
@curiousercreative I squashed your commits and edited the changelog to make it shorter (people reading the changelog can click the links to the issue and PR to read more). The commit still shows you as the author. Thanks for the contribution.

@micbar please get the appropriate developer to review this. And I guess @davitol might have some access already to a test Google Drive where he can manually reproduce the problem and test the fix.

**Comment by phil-davis** [comment]
Note: minor codecov fail because this PR changes GoogleDrive-specific code that will not be covered by unit tests.

File: apps/files_external/lib/Lib/Storage/Google.php [hunk 1]
```diff
@@ -286,7 +286,11 @@ public function opendir($path) {
 						if ($child->getMimeType() === self::MAP) {
 							continue; // No method known to transfer map files, ignore it
 						} elseif ($child->getMimeType() !== self::FOLDER) {
-							$name .= '.'.$this->getGoogleDocExtension($child->getMimeType());
+							$extension = $this->getGoogleDocExtension($child->getMimeType());
+							// don't append an empty extension as they will create broken paths
+							if ($extension != '') {
```
Comment (jvillafanez) [inline apps/files_external/lib/Lib/Storage/Google.php]: Use `!==` by default unless there is a reason not to do so (which should be explained in a comment)

File: apps/files_external/lib/Lib/Storage/Google.php [hunk 2]
```diff
@@ -286,7 +286,11 @@ public function opendir($path) {
...
+								$name .= '.'.$extension;
```
Comment (jvillafanez) [inline apps/files_external/lib/Lib/Storage/Google.php]:
```
$name .= ".{$extension}";
```
```
$name .= '.' . $extension;
```
I'm more in favor of the first one as a general approach, although in this case it doesn't matter too much. You can include spaces between the `.` operator. Up to you

**Comment by davitol** [comment]
> And I guess @davitol might have some access already to a test Google Drive where he can manually reproduce the problem and test the fix.

Reproduced and tested the fix. With the fix dangling dot disappears and it is possible to download the file.

**Review by jvillafanez — APPROVED** [review]

**Pull request merged**